dali/dfu/dfuwu.cpp

```diff
@@ -2972,6 +3135,34 @@ class CDFUWorkUnitFactory : implements IDFUWorkUnitFactory, implements ISDSSubsc
         // created time stamp? TBD
         return ret;
     }
+
+    IDFUWorkUnit * createPublisherWorkUnit() override
+    {
+        StringBuffer wuid;
+        newWUID(wuid, 'P');
+        StringBuffer wuRoot;
+        getXPath(wuRoot, wuid.str());
+        IRemoteConnection *conn = querySDS().connect(wuRoot.str(), myProcessSession(), RTM_LOCK_WRITE|RTM_CREATE_UNIQUE, SDS_LOCK_TIMEOUT);
+        conn->queryRoot()->setProp("@xmlns:xsi", "http://www.w3.org/1999/XMLSchema-instance");
+        IDFUWorkUnit *ret = new CDFUWorkUnit(this, conn, NULL, true);
+        ret->setCommand(DFUcmd_publish);
+        return ret;
+    }
+
+    IDFUWorkUnit * createPublisherSubTask(IDFUWorkUnit *publisherWu) override
+    {
+        IDFUprogress *progress = publisherWu->queryUpdateProgress();
+        unsigned taskId = progress->incPublisherTaskCount();
+
+        StringBuffer wuRoot;
+        getXPath(wuRoot, publisherWu->queryId(), taskId);
+        IRemoteConnection* conn = querySDS().connect(wuRoot.str(), myProcessSession(), RTM_LOCK_WRITE|RTM_CREATE_UNIQUE, SDS_LOCK_TIMEOUT);
+        conn->queryRoot()->setProp("@xmlns:xsi", "http://www.w3.org/1999/XMLSchema-instance");
+        IDFUWorkUnit *ret = new CDFUWorkUnit(this, conn, NULL, true);
```

Comment:
minor: safer to make owned, then return via getClear()

I'd also own 'conn' and getClear() when passing ownership to constructor